### PR TITLE
Enable OpenSplat on AMD Radeon R9700 graphic card with ROCm 7.0.0 and libtorch 2.5.1

### PR DESCRIPTION
**Pull request by charyang-ai**
Porting the gaussian algorithm with ROCm 7.0.0 and Torch 2.5.1. 
It's verified on AMD Radeon R9700 with RDNA4 arch. The training loss could converage with this new configuration which will failed with ROCm <=6.4.x.
image

### CODE REVIEW

**Comment by charyang-ai** [comment]
The gpu utilization status shows as below:
image

**Comment by pierotofy** [comment]
Looks good to me, thanks @charyang-ai !
